src/CMakeLists.txt

```diff
@@ -24,6 +24,12 @@ file(GLOB SOURCE_CPP
     "*.cpp"
 )
 
+if(QT_VERSION_MAJOR EQUAL 6)
+    list(FILTER SOURCE_CPP EXCLUDE REGEX "SBarcodeFilter.")
+else()
+    list(FILTER SOURCE_CPP EXCLUDE REGEX "SBarcodeScanner.")
+endif()
+
```

Comment:
I'm still not sure about this - kinda looks like a hack.
Should we delete SBarcodeFilter from Qt6 version even if it's extended by `SBarcodeScanner`?
Maybe we should just add a subfolder like `qt6` to keep the additional classes for Qt6

Reply:
The idea of excluding the related class according to Qt version is prevent to get error from not existing libraries when you compile the project for Qt6(e.g. QVideoFilterRunnable, QAbstractVideoFilter). Likewise in Qt5 for QVideoSink. Instead of putting #ifdef directives to everywhere it will be good to do this in one place.